build.sbt

```diff
@@ -1471,6 +1460,7 @@ lazy val lmCoursierDependencies = Def.settings(
     "io.get-coursier.jniutils" % "windows-jni-utils-lmcoursier" % jniUtilsVersion,
     "net.hamnaberg" %% "dataclass-annotation" % dataclassScalafixVersion % Provided,
     "org.scalatest" %% "scalatest" % "3.2.19" % Test,
+    Dependencies.scalaCollectionCompat,
```

Comment:
Need this otherwise we get hit with

```
[info] lmcoursier.CoursierDependencyResolutionTests *** ABORTED ***
[info]   java.lang.NoClassDefFoundError: scala/collection/compat/immutable/package$
[info]   at coursier.core.Resolution.helper$4(Resolution.scala:1291)
[info]   at coursier.core.Resolution.orderedDependencies(Resolution.scala:1312)
[info]   at coursier.core.Resolution.dependencyArtifacts(Resolution.scala:1350)
[info]   at coursier.Artifacts$.artifacts(Artifacts.scala:260)
[info]   at coursier.Artifacts.$anonfun$ioResult$1(Artifacts.scala:79)
[info]   at scala.collection.immutable.List.flatMap(List.scala:294)
[info]   at scala.collection.immutable.List.flatMap(List.scala:79)
[info]   at coursier.Artifacts.ioResult(Artifacts.scala:73)
[info]   at coursier.Artifacts$ArtifactsTaskOps$.eitherResult$extension(Artifacts.scala:207)
[info]   at lmcoursier.internal.ArtifactsRun$.result(ArtifactsRun.scala:54)
[info]   ...
[info]   Cause: java.lang.ClassNotFoundException: scala.collection.compat.immutable.package$
[info]   at coursier.core.Resolution.helper$4(Resolution.scala:1291)
[info]   at coursier.core.Resolution.orderedDependencies(Resolution.scala:1312)
[info]   at coursier.core.Resolution.dependencyArtifacts(Resolution.scala:1350)
[info]   at coursier.Artifacts$.artifacts(Artifacts.scala:260)
[info]   at coursier.Artifacts.$anonfun$ioResult$1(Artifacts.scala:79)
[info]   at scala.collection.immutable.List.flatMap(List.scala:294)
[info]   at scala.collection.immutable.List.flatMap(List.scala:79)
[info]   at coursier.Artifacts.ioResult(Artifacts.scala:73)
[info]   at coursier.Artifacts$ArtifactsTaskOps$.eitherResult$extension(Artifacts.scala:207)
[info]   at lmcoursier.internal.ArtifactsRun$.result(ArtifactsRun.scala:54)
```

c.c. https://github.com/Friendseeker/sbt/actions/runs/11298101112/job/31426345865

Reply:
https://github.com/coursier/coursier/pull/3095 should resolve it. In the meantime we shall keep the workaround.

Reply:
Never mind it is actually because of the exclusion rule below...